src/parse.js

```diff
@@ -267,9 +272,37 @@ function parse(source, root, options) {
         // Otherwise the meaning is ambiguous between proto2 and proto3
         root.setOption("syntax", syntax);
 
+        if (isProto3) {
+            for (var proto3Key of Object.keys(proto3Defaults)) {
+                setParsedOption(root, proto3Key, proto3Defaults[proto3Key]);
+            }
+        } else {
+            for (var proto2Key of Object.keys(proto2Defaults)) {
+                setParsedOption(root, proto2Key, proto2Defaults[proto2Key]);
+            }
+        }
+
         skip(";");
     }
 
+    function parseEdition() {
+        skip("=");
+        edition = readString();
+        const supportedEditions = ["2023"];
+
+        /* istanbul ignore if */
+        if (!supportedEditions.includes(edition))
+            throw illegal(edition, "edition");
+
+        root.setOption("edition", edition);
```

Comment:
It's just following the pattern for setting syntax. I assume it will be useful later on when we have to decide how to treat things based on proto2, proto3 and editions.